crates/kernel/src/tasks/mod.rs

```diff
@@ -162,3 +162,79 @@ pub mod vm_test_utils {
         })
     }
 }
+
+pub mod scheduler_test_utils {
+    use crate::config::Config;
+    use crate::tasks::sessions::Session;
+    use crate::vm::UncaughtException;
+    use moor_db::Database;
+    use moor_values::model::{CommandError, WorldStateSource};
+    use moor_values::var::{Error::E_VERBNF, Objid, Var};
+    use std::sync::Arc;
+
+    use super::scheduler::{Scheduler, SchedulerError, TaskWaiterResult};
+    use super::TaskId;
+    use crate::tasks::scheduler_test_utils::SchedulerError::{
+        CommandExecutionError, TaskAbortedException,
+    };
+
+    pub type ExecResult = Result<Var, UncaughtException>;
+
+    fn execute<F>(database: Arc<dyn Database + Send + Sync>, fun: F) -> Result<Var, SchedulerError>
+    where
+        F: FnOnce(Arc<dyn WorldStateSource>, Arc<Scheduler>) -> Result<TaskId, SchedulerError>,
+    {
+        let scheduler = Arc::new(Scheduler::new(database.clone(), Config::default()));
+        let task_id = fun(database.world_state_source().unwrap(), scheduler.clone())?;
+        let subscriber = scheduler.subscribe_to_task(task_id).unwrap();
+
+        let loop_scheduler = scheduler.clone();
+        let scheduler_loop_jh = std::thread::Builder::new()
+            .name("moor-scheduler".to_string())
+            .spawn(move || loop_scheduler.run())
+            .unwrap();
+
+        let result = match subscriber
+            .recv()
+            .inspect_err(|e| eprintln!("subscriber.recv() failed: {e}"))
+            .unwrap()
+        {
+            TaskWaiterResult::Error(TaskAbortedException(UncaughtException { code, .. })) => {
+                Ok(code.into())
+            }
+            TaskWaiterResult::Error(CommandExecutionError(CommandError::NoCommandMatch)) => {
+                Ok(E_VERBNF.into())
```

Comment:
This isn't considered an Error?

Reply:
Abstractly: not as far as scheduling goes! Note the error return type from `execute` is `SchedulerError`.

Concretely: the result of this function is compared to the expected value defined in a `.moot` test file; expected values are written as normal MOO expressions. The way to express "verb not found" with a MOO value is `E_VERBNF`, and so we must return `E_VERBNF` here to be able to express this is the expected behavior of the test. You can see this in action in the first block of `huh.moot`.

Reply:
... possibly this is not the ideal layer to express this? Maybe it should be a layer up? /shrug I don't care *too* much about the architectural correctness of test helpers (until they become a problem)

Reply:
I think just documenting that this particular MOO E_ value is OK and should be tested in the moot, while the others are system level errors is fine.